css/css-display/accessibility/display-contents-role-and-label.html

```diff
@@ -0,0 +1,290 @@
+          <div role="row">
+              <div role="columnheader" style="display: contents;" data-expectedrole="columnheader" data-expectedlabel="x" data-testname="div with role columnheader with display: contents, in div with role rowgroup within div with table role both with display: contents, has columnheader role" class="ex-role-and-label">x</div>
+              <div role="columnheader">x</div>
+          </div>
+      </div>
+      <div role="rowgroup">
+          <div role="row">
+              <div role="cell">x</div>
+              <div role="cell">x</div>
+          </div>
+      </div>
+  </div>
+
+  <table role="grid" style="display: contents;" data-expectedrole="grid" data-testname="table with role grid and display: contents has grid role">
+      <thead>
+        <tr style="display: contents;" data-expectedrole="row" data-testname="tr within table with role grid, both with display: contents, has row role">
+          <th style="display: contents;" data-expectedrole="columnheader" data-expectedlabel="x" data-testname="th within table with role grid, both with display: contents, has columnheader role" class="ex-role-and-label">x</th>
+          <th>x</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td style="display: contents;" data-expectedrole="gridcell" data-expectedlabel="x" data-testname="td within table with role grid, both with display: contents, has gridcell role" class="ex-role-and-label">x</td>
+          <td>x</td>
+        </tr>
+      </tbody>
+    </table>
+
+  <div role="grid" style="display: contents;" data-expectedrole="grid" data-testname="div with role grid with display: contents has grid role" class="ex-role">
+    <div role="rowgroup" style="display: contents;" data-expectedrole="rowgroup" data-testname="div with role rowgroup in div with role grid, both with display: contents, has rowgroup role" class="ex-role">
+        <div role="row">
+            <div role="columnheader" style="display: contents;" data-expectedrole="columnheader" data-expectedlabel="x" data-testname="div with role columnheader with display: contents, in div with role rowgroup within div with grid role both with display: contents, has rowgroup role" class="ex-role-and-label">x</div>
+            <div role="columnheader">x</div>
+        </div>
+    </div>
+    <div role="rowgroup">
+        <div role="row">
+            <div role="gridcell"></div>
+            <div role="gridcell" style="display: contents;" data-expectedrole="gridcell" data-expectedlabel="x" data-testname="div with gridcell role and display: contents, within div with role row contained in a div with role grid both with display: contents, has gridcell role" class="ex-role-and-label">x</div>
+        </div>
+    </div>
+  </div>
+
+  <div role="grid" tabindex="0" style="display: contents;" data-expectedrole="grid" data-testname="div with role grid, tabindex=0 and display: contents has grid role" class="ex-role">
+  </div>
```

Comment:
Isn't this invalid? Why do you expect a grid role here even though there are no contents?

Reply:
It should be valid to test interactivity of the `grid` wrapper 🙂. We added this test from your feedback here: https://github.com/web-platform-tests/wpt/pull/43740#discussion_r1442407641/

Reply:
I recall the feedback, but thought that was in the context of a well-formed grid, rather than an empty grid. As-is, it's unclear to me what the returned `computedrole` should be for a single `<div role="grid" tabindex="0">` with no row or cell contents. In theory, either `grid` or some other non-`generic` container role (like `group`) might be valid, unless I've missed a UA requirement.